### DISCUSSION TITLE
OT-generated filenames

### DISCUSSION THREAD

**Discussion by markgif** [dita-ot/dita-ot#4413] [Q&A] [answered]
I am just verifying something I have observed in using OT 4.1.2, that I could not find any docs on. I am doing normal PDF output.

The OT will generate filenames like `5d53d743428dbc99192bbe7b840acb7ed193713f.dita` for DITA files, and that file will be on the file system. This I expected. I assume it provides flexibility.

But I noticed it is different with graphics files. A graphics file like `test6in0327141035.svg` gets a generated filename like `9df41a48c2d8613d864f3619d891e392cbf71bdf.svg`, but no copy is made on the file system. The generated name appears in a few temp files (`xx_MERGED.xml`, `stage1.xml`) as it goes through the pipeline. But when it creates `topic.fo` right before the PDF is made, the OT reverts to the original actual filename. It looks like the mapping between the real and generated names is in the `.job.xml` file.

Is this roughly right? I noticed this while updating an old hacked-up OT to v4.1.2.

**Comment by jelovirt** [comment, marked as answer]
Yes, that's correct. The map-first pre-processing used by PDF uses UUIDs for file names. Images, however, are not copied to the temporary directory because pre-processing doesn't need to modify them. To make the code more simple, UUIDs are used everywhere.

**Comment by markgif** [comment]
Thanks @jelovirt good to know.
